tests/test_vmu.py

```diff
@@ -0,0 +1,192 @@
+                                    'PackageSet label or packages equality broken')
+        self.assertEqualWithResults(vmu.PackageSet('foo', [1, 2, 3], True, False),
+                                    vmu.PackageSet('foo', [1, 2, 3], False, False),
+                                    'PackageSet SELinux equality broken')
+        # Test mismatched tags
+        self.assertRaises(vmu.ParamError, vmu.PackageSet.__eq__,
+                          vmu.PackageSet('foo', [1, 2, 3], True, False),
+                          vmu.PackageSet('bar', [1, 2, 3], False, False))
+
+    def test_inequality(self):
+        self.assertNotEqualWithResults(vmu.PackageSet('foo', [1, 2, 3]),
+                                       vmu.PackageSet('bar', [4, 5, 6]),
+                                       'PackageSet packages inequality broken')
+        self.assertNotEqualWithResults(vmu.PackageSet('foo', [1, 2, 3]),
+                                       vmu.PackageSet('foo', [4, 5, 6]),
+                                       'PackageSet packages inequality broken')
+        self.assertNotEqualWithResults(vmu.PackageSet('foo', [1, 2, 3], java=True),
+                                       vmu.PackageSet('foo', [1, 2, 3], java=False),
+                                       'PackageSet java inequality broken')
+        # Test mismatched tags
+        self.assertRaises(vmu.ParamError, vmu.PackageSet.__eq__,
+                          vmu.PackageSet('foo', [1, 2, 3], True, False),
+                          vmu.PackageSet('bar', [1, 2, 3], False, False))
+
+    def test_sorting(self):
+        unsorted_pkgs = [vmu.PackageSet('All + GRAM', [1]),
+                vmu.PackageSet('GridFTP', [1]),
+                vmu.PackageSet('VOMS', [1]),
+                vmu.PackageSet('All + GRAM (3.2)', [1]),
+                vmu.PackageSet('All', [1]),
+                vmu.PackageSet('HTCondor', [1]),
+                vmu.PackageSet('GUMS', [1]),
+                vmu.PackageSet('BeStMan', [1]),
+                vmu.PackageSet('foo', [1])]
+        unsorted_pkgs.sort()
+        self.assertEqualWithResults(vmu.PackageSet.LABEL_ORDER + ['foo'],
+                                    [x.label for x in unsorted_pkgs],
+                                    'Unexpected sort order')
+
+    def test_defaults(self):
+        pkg_set = vmu.PackageSet('foo', [1, 2, 3])
+        self.assertEqual(pkg_set.selinux, vmu.PackageSet.SELINUX_DEFAULT, 'Failed to set SELinux default')
+        self.assertEqual(pkg_set.java, vmu.PackageSet.OSG_JAVA_DEFAULT, 'Failed to set OSG Java default')
+
+
+    def test_hashable(self):
+        selinux_enabled = vmu.PackageSet('foo', [1, 2, 3], True, True)
+        self.assert_(set([selinux_enabled]),
+                     'PackageSet.__hash__() broken')
+        self.assertEqualWithResults(set([selinux_enabled]),
+                                    set([selinux_enabled, selinux_enabled]),
+                                    'PackageSet hash equality')
```

Comment:
I don't get what the second assert is doing. Also, shouldn't you test if a PackageSet with SELinux enabled hashes the same as a PackageSet with SELinux disabled?